### DISCUSSION TITLE
open closed drawer state, how?

### DISCUSSION THREAD

**Discussion by tombohub** [remix-run/remix#9317] [Q&A] [answered]
In react I am managing open closed drawer state with redux:

```javascript
function MenuButton() {
  const dispatch = useAppDispatch();

  return <Burger onClick={() => dispatch(layoutActions.toggleDrawer())} />;
}
```

how do you do it in remix globally (without passing useState around)? Do you use search param `?drawer=open` ?

**Comment by sergiodxa** [comment, marked as answer]
You could use useState and a context for that, although you can probably put that drawer in a parent route and keep the state, toggle button and drawer element all in one place.
